test/webgl-utils/track-context.spec.js

```diff
@@ -1,78 +1,155 @@
 import {createTestContext} from '../setup';
 import {GL} from 'luma.gl';
-import trackContext, {pushContextState, popContextState} from '../../src/webgl-utils/track-context';
-// import {trackContext, TEST_EXPORTS} from '../../src/webgl///';
-// import {getParameter, setParameter, withParameters, resetParameters} from 'luma.gl';
-
+import trackContext, {pushContextState, popContextState, setParameters}
+  from '../../src/webgl-utils/track-context';
+import {glGetParameter} from '../../src/webgl-utils/parameters';
 import test from 'tape-catch';
 
 // Settings test, don't reuse a context
 const fixture = {
   gl: createTestContext({debug: true})
 };
 
-import GL_PARAMETERS from '../../src/webgl/api/parameters';
+import {GL_PARAMETER_DEFAULTS} from '../../src/webgl-utils/parameters';
 
 // eslint-disable-next-line
-const SETTINGS = {
-  blend: false,
-  blendColor: [0, 0, 0, 0],
-  blendEquation: [GL.FUNC_ADD, GL.FUNC_ADD], // [GL.BLEND_EQUATION_RGB, GL.BLEND_EQUATION_ALPHA],
-  blendFunc: [GL.ONE, GL.ZERO, GL.ONE, GL.ZERO],
-
-  clearColor: [0, 0, 0, 0],
-  colorMask: [true, true, true, true],
-  // colorWritemask: ,
-
-  cullFace: false,
-  cullFaceMode: GL.BACK,
-
-  depthTest: false,
-  depthClearValue: 1,
-  depthFunc: GL.LESS,
-  depthRange: [0, 1],
-  depthWritemask: true,
-
-  dither: true,
-
-  frontFace: GL.CCW,
-
-  generateMipmapHint: GL.DONT_CARE,
-
-  lineWidth: 1,
-
-  polygonOffsetFill: false,
-  polygonOffset: [0, 0],
-
-  // sampleCoverage: GL.SAMPLE_COVERAGE,
-
-  scissorTest: false,
-  scissorBox: [0, 0, 1024, 1024],
-
-  stencilTest: false,
-  stencilClearValue: 0,
-  stencilMask: [0xFFFFFFFF, 0xFFFFFFFF],
-  stencilFunc: [GL.ALWAYS, 0, 0xFFFFFFFF, GL.ALWAYS, 0, 0xFFFFFFFF],
-  stencilOp: [GL.KEEP, GL.KEEP, GL.KEEP, GL.KEEP, GL.KEEP, GL.KEEP],
-
-  viewport: [0, 0, 1024, 1024],
+const GL_PARAMETER_SETTINGS_ONE = {
```

Comment:
One thing we are not testing here is "raw WebGL calls"...

e.g. `gl.viewport`, that it gets intercepted correctly.